### PR TITLE
Net6 sdk

### LINKED ISSUE #272: Enforce coding standard more strictly
Increase severity of "suggestion" and "none" to "warning". See https://github.com/nunit/nunit.analyzers/pull/264#discussion_r460432710 and the rest of the thread for context.

### PR DESCRIPTION
**Pull request by manfred-brands**
Fixes #272 

- Upgrade to .NET6 SDK
- Move common definitions from individual .csproj files to Directory.Build.props
- Fix nullability issues.
- Do what we preach, reference nunit.analyzers as an Analyzer
- Convert suggestions for code style to warnings. The code was already clean, but this way we keep it that way.

@mikkelbu this is in preparation of #416 where we need way more projects, one for every supported compiler, to prevent duplication, I moved to using the Directory.Build.props to host common definitions.

### CODE REVIEW

**Comment by mikkelbu** [comment]
Thanks for doing this @manfred-brands. I think I'll have time to review it tonight or tomorrow.

File: src/nunit.analyzers.tests/TestMethodUsage/TestMethodUsageAnalyzerTests.cs [hunk 1]
```diff
@@ -46,22 +46,28 @@ public void VerifySupportedDiagnostics()
...
-                    $"{diagnostic.Id} : {nameof(DiagnosticDescriptor.Title)}");
-                Assert.That(diagnostic.Category, Is.EqualTo(Categories.Structure),
-                    $"{diagnostic.Id} : {nameof(DiagnosticDescriptor.Category)}");
-                Assert.That(diagnostic.DefaultSeverity, Is.EqualTo(DiagnosticSeverity.Error),
-                    $"{diagnostic.Id} : {nameof(DiagnosticDescriptor.DefaultSeverity)}");
-            }
+                foreach (var diagnostic in diagnostics)
+                {
+                    Assert.That(diagnostic.Title.ToString(CultureInfo.InvariantCulture), Is.Not.Empty,
+  $"{diagnostic.Id} : {nameof(DiagnosticDescriptor.Title)}");
```
Comment (mikkelbu) [inline src/nunit.analyzers.tests/TestMethodUsage/TestMethodUsageAnalyzerTests.cs:54]:
```suggestion
                        $"{diagnostic.Id} : {nameof(DiagnosticDescriptor.Title)}");
```

File: src/nunit.analyzers.tests/TestMethodUsage/TestMethodUsageAnalyzerTests.cs [hunk 2]
```diff
@@ -46,22 +46,28 @@ public void VerifySupportedDiagnostics()
...
+                }
+            });
 
             var diagnosticMessage = diagnostics.Select(_ => _.MessageFormat.ToString(CultureInfo.InvariantCulture)).ToImmutableArray();
 
-            Assert.That(diagnosticMessage, Contains.Item(TestMethodUsageAnalyzerConstants.ExpectedResultTypeMismatchMessage),
+            Assert.Multiple(() =>
+            {
+                Assert.That(diagnosticMessage, Contains.Item(TestMethodUsageAnalyzerConstants.ExpectedResultTypeMismatchMessage),
                 $"{TestMethodUsageAnalyzerConstants.ExpectedResultTypeMismatchMessage} is missing.");
```
Comment (mikkelbu) [inline src/nunit.analyzers.tests/TestMethodUsage/TestMethodUsageAnalyzerTests.cs:67]:
```suggestion
                    $"{TestMethodUsageAnalyzerConstants.ExpectedResultTypeMismatchMessage} is missing.");
```

File: src/nunit.analyzers.tests/Extensions/ITypeSymbolExtensionsTests.cs [hunk 3]
```diff
@@ -208,9 +208,12 @@ private static async Task<ImmutableArray<ITypeSymbol>> GetTypeSymbolAsync(string
 
             foreach (var typeName in typeNames)
             {
-                types.Add(rootAndModel.Model.GetDeclaredSymbol(rootAndModel.Node
-                    .DescendantNodes().OfType<TypeDeclarationSyntax>()
-                    .Where(_ => _.Identifier.ValueText == typeName).Single()));
+                INamedTypeSymbol? item = rootAndModel.Model.GetDeclaredSymbol(
+                    rootAndModel.Node
+                                .DescendantNodes().OfType<TypeDeclarationSyntax>()
+                                .Where(_ => _.Identifier.ValueText == typeName).Single());
```
Comment (mikkelbu) [inline src/nunit.analyzers.tests/Extensions/ITypeSymbolExtensionsTests.cs:214]: I was wondering if we should move this into a variable for readability. So something like

```
TypeDeclarationSyntax typeDeclaration = rootAndModel.Node
    .DescendantNodes()
    .OfType<TypeDeclarationSyntax>()
    .Where(_ => _.Identifier.ValueText == typeName)
    .Single();
INamedTypeSymbol? item = rootAndModel.Model.GetDeclaredSymbol(typeDeclaration);
...
```

**Review by mikkelbu — APPROVED** [review]
Looks great @manfred-brands 👍 . I only have the smallest of comments, but I'm also happy to merge it as it is.

File: src/nunit.analyzers.tests/TestMethodUsage/TestMethodUsageAnalyzerTests.cs (hunk 1 above)
Comment (manfred-brands) [reply src/nunit.analyzers.tests/TestMethodUsage/TestMethodUsageAnalyzerTests.cs]: Done.

File: src/nunit.analyzers.tests/TestMethodUsage/TestMethodUsageAnalyzerTests.cs (hunk 2 above)
Comment (manfred-brands) [reply src/nunit.analyzers.tests/TestMethodUsage/TestMethodUsageAnalyzerTests.cs]: Done.

File: src/nunit.analyzers.tests/Extensions/ITypeSymbolExtensionsTests.cs (hunk 3 above)
Comment (manfred-brands) [reply src/nunit.analyzers.tests/Extensions/ITypeSymbolExtensionsTests.cs]: This code was more or less in there from the beginning, but you are right; splitting is make it more readable.
Done.

**Comment by manfred-brands** [comment]
Thanks @mikkelbu for your review. Changes applied, ready for merge.

**Comment by mikkelbu** [comment]
Thanks for the PR @manfred-brands. I'm merging it now.

**Comment by manfred-brands** [comment]
@mikkelbu This might have broken your versioning of the dlls which I detected when working on #416 
I suggest to delay any release until then.
There is only one bugfix in since the last version and they should give a reason when using `Assert.Ignore()` and `Assert.Fail()`.

**Comment by mikkelbu** [comment]
@manfred-brands Thanks for noticing this. I should have given it some thought when I performed the review

**Comment by manfred-brands** [comment]
@mikkelbu This was because I removed AssemblyInfo.cs files which were used in build.cake to add Generated.AssemblyInfo.cs with version information.
This is fixed with #439 which set the Version parameter as part of the DotNetBuild step.

**Comment by manfred-brands** [comment]
We should also get #438 into that new release.

**Pull request merged**